hw7-1/dict.c

```diff
@@ -0,0 +1,241 @@
+    if (preCurr->left->value == NULL) {
+        return memoryAllocationError;
+    }
+    strcpy(preCurr->left->value, value);
+    preCurr->left->key = key;
+    return OK;
+}
+
+Error delete(Dictionary *dictionary, int key) {
+    if (dictionary == NULL) {
+        return noDict;
+    }
+    bool isFound = false;
+    depthSearch(dictionary->root, &isFound, key);
+    if (!isFound) {
+        return OK;
+    }
+    DictionaryElement *elementForDelete = dictionary->root;
+    DictionaryElement *prevElementForDelete = NULL;
+    bool rightWay = true; //right or left
+    while (elementForDelete->key != key) {
+        prevElementForDelete = elementForDelete;
+        if (elementForDelete->key > key) {
+            elementForDelete = elementForDelete->left;
+            rightWay = false;
+        } else {
+            elementForDelete = elementForDelete->right;
+            rightWay = true;
+        }
+    }
+    if (elementForDelete->right == NULL && elementForDelete->left == NULL) {
+        if (dictionary->root == elementForDelete) {
+            free(elementForDelete);
+            dictionary->root = NULL;
+            return OK;
+        }
+        if (rightWay) {
+            prevElementForDelete->right = NULL;
+        } else {
+            prevElementForDelete->left = NULL;
+        }
+        free(elementForDelete);
+        return OK;
+    }
+    if (elementForDelete->left == NULL) {
+        if (prevElementForDelete == NULL) {
+            dictionary->root = dictionary->root->right;
+        } else if (rightWay) {
+            prevElementForDelete->right = elementForDelete->right;
+        } else {
+            prevElementForDelete->left = elementForDelete->right;
+        }
+        free(elementForDelete->value);
+        free(elementForDelete);
+        return OK;
+    }
+    DictionaryElement *elementForChangeWithDelete = elementForDelete;
+    prevElementForDelete = elementForDelete;
+    elementForDelete = elementForDelete->left;
+    rightWay = false;
+    while (elementForDelete->right != NULL) {
+        prevElementForDelete = elementForDelete;
+        elementForDelete = elementForDelete->right;
+        rightWay = true;
+    }
+    elementForChangeWithDelete->key = elementForDelete->key;
+    free(elementForChangeWithDelete->value);
+    elementForChangeWithDelete->value = calloc(1, sizeof(elementForDelete->value));
+    if (elementForChangeWithDelete->value == NULL) {
+        return memoryAllocationError;
+    }
+    strcpy(elementForChangeWithDelete->value, elementForDelete->value);
+    if (rightWay) {
+        prevElementForDelete->right = elementForDelete->left;
+    } else {
+        prevElementForDelete->left = elementForDelete->left;
+    }
+    free(elementForDelete->value);
+    free(elementForDelete);
+    return OK;
+}
+
+void printDict(Dictionary *dictionary) {
+    printf("\nYour dictionary:\n");
+    depthPrint(dictionary->root);
+}
+
+void deleteDictDepth(DictionaryElement *dictionaryElement) {

```

Comment:
```suggestion
void deleteDictNodes(DictionaryElement *dictionaryElement) {
```
Мы же не глубину удаляем, а узлы дерева